{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ff42e823",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyautogui in c:\\users\\user\\anaconda3\\lib\\site-packages (0.9.53)\n",
      "Requirement already satisfied: pygetwindow>=0.0.5 in c:\\users\\user\\anaconda3\\lib\\site-packages (from pyautogui) (0.0.9)\n",
      "Requirement already satisfied: mouseinfo in c:\\users\\user\\anaconda3\\lib\\site-packages (from pyautogui) (0.1.3)\n",
      "Requirement already satisfied: pymsgbox in c:\\users\\user\\anaconda3\\lib\\site-packages (from pyautogui) (1.0.9)\n",
      "Requirement already satisfied: PyTweening>=1.0.1 in c:\\users\\user\\anaconda3\\lib\\site-packages (from pyautogui) (1.0.4)\n",
      "Requirement already satisfied: pyscreeze>=0.1.21 in c:\\users\\user\\anaconda3\\lib\\site-packages (from pyautogui) (0.1.28)\n",
      "Requirement already satisfied: pyrect in c:\\users\\user\\anaconda3\\lib\\site-packages (from pygetwindow>=0.0.5->pyautogui) (0.2.0)\n",
      "Requirement already satisfied: pyperclip in c:\\users\\user\\anaconda3\\lib\\site-packages (from mouseinfo->pyautogui) (1.8.2)\n"
     ]
    }
   ],
   "source": [
    "!pip install pyautogui"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5a594e33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyperclip in c:\\users\\user\\anaconda3\\lib\\site-packages (1.8.2)\n"
     ]
    }
   ],
   "source": [
    "!pip install pyperclip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7230516f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting schedule\n",
      "  Downloading schedule-1.1.0-py2.py3-none-any.whl (10 kB)\n",
      "Installing collected packages: schedule\n",
      "Successfully installed schedule-1.1.0\n"
     ]
    }
   ],
   "source": [
    "!pip install schedule"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3094e085",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '1.png'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[1;32mIn [7]\u001b[0m, in \u001b[0;36m<cell line: 9>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      6\u001b[0m \u001b[38;5;66;03m# 경로를 .py파일의 실행경로로 이동, 현재 경로로 이동\u001b[39;00m\n\u001b[0;32m      7\u001b[0m os\u001b[38;5;241m.\u001b[39mchdir(os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mdirname(os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mabspath(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mautokakao.ipynb\u001b[39m\u001b[38;5;124m'\u001b[39m)))\n\u001b[1;32m----> 9\u001b[0m picPosition \u001b[38;5;241m=\u001b[39m \u001b[43mpyautogui\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlocateOnScreen\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m1.png\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m     10\u001b[0m \u001b[38;5;28mprint\u001b[39m(picPosition)\n\u001b[0;32m     12\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m picPosition \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py:175\u001b[0m, in \u001b[0;36mraisePyAutoGUIImageNotFoundException.<locals>.wrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    172\u001b[0m \u001b[38;5;129m@functools\u001b[39m\u001b[38;5;241m.\u001b[39mwraps(wrappedFunction)\n\u001b[0;32m    173\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapper\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m    174\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 175\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m wrappedFunction(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    176\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m pyscreeze\u001b[38;5;241m.\u001b[39mImageNotFoundException:\n\u001b[0;32m    177\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ImageNotFoundException\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py:213\u001b[0m, in \u001b[0;36mlocateOnScreen\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    211\u001b[0m \u001b[38;5;129m@raisePyAutoGUIImageNotFoundException\u001b[39m\n\u001b[0;32m    212\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mlocateOnScreen\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m--> 213\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m pyscreeze\u001b[38;5;241m.\u001b[39mlocateOnScreen(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py:373\u001b[0m, in \u001b[0;36mlocateOnScreen\u001b[1;34m(image, minSearchTime, **kwargs)\u001b[0m\n\u001b[0;32m    371\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    372\u001b[0m     screenshotIm \u001b[38;5;241m=\u001b[39m screenshot(region\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;66;03m# the locateAll() function must handle cropping to return accurate coordinates, so don't pass a region here.\u001b[39;00m\n\u001b[1;32m--> 373\u001b[0m     retVal \u001b[38;5;241m=\u001b[39m locate(image, screenshotIm, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    374\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    375\u001b[0m         screenshotIm\u001b[38;5;241m.\u001b[39mfp\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py:353\u001b[0m, in \u001b[0;36mlocate\u001b[1;34m(needleImage, haystackImage, **kwargs)\u001b[0m\n\u001b[0;32m    351\u001b[0m \u001b[38;5;66;03m# Note: The gymnastics in this function is because we want to make sure to exhaust the iterator so that the needle and haystack files are closed in locateAll.\u001b[39;00m\n\u001b[0;32m    352\u001b[0m kwargs[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mlimit\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m--> 353\u001b[0m points \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mtuple\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mlocateAll\u001b[49m\u001b[43m(\u001b[49m\u001b[43mneedleImage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhaystackImage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    354\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(points) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[0;32m    355\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m points[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py:262\u001b[0m, in \u001b[0;36m_locateAll_python\u001b[1;34m(needleImage, haystackImage, grayscale, limit, region, step, confidence)\u001b[0m\n\u001b[0;32m    259\u001b[0m needleFileObj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    260\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(needleImage, (\u001b[38;5;28mstr\u001b[39m, unicode)):\n\u001b[0;32m    261\u001b[0m     \u001b[38;5;66;03m# 'image' is a filename, load the Image object\u001b[39;00m\n\u001b[1;32m--> 262\u001b[0m     needleFileObj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mneedleImage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mrb\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m    263\u001b[0m     needleImage \u001b[38;5;241m=\u001b[39m Image\u001b[38;5;241m.\u001b[39mopen(needleFileObj)\n\u001b[0;32m    265\u001b[0m haystackFileObj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '1.png'"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-108:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-109:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-110:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-111:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-112:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-113:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-114:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-115:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-116:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-117:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-118:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-119:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-120:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-121:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-122:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-123:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Box(left=1526, top=102, width=153, height=64)\n",
      "None\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-125:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-126:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n",
      "Exception in thread Thread-127:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Box(left=1526, top=102, width=153, height=64)\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-129:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1525, top=100, width=158, height=69)\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-130:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-132:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1526, top=102, width=153, height=64)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-133:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Box(left=1526, top=102, width=153, height=64)\n",
      "None\n",
      "None\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-136:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 23, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 582, in center\n",
      "    return Point(coords[0] + int(coords[2] / 2), coords[1] + int(coords[3] / 2))\n",
      "TypeError: 'NoneType' object is not subscriptable\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n"
     ]
    }
   ],
   "source": [
    "import pyautogui\n",
    "import pyperclip\n",
    "import time\n",
    "import os\n",
    "\n",
    "# 경로를 지정한 파일의 실행경로로 이동, 그리고 절대경로로 지정한다.\n",
    "os.chdir(os.path.dirname(os.path.abspath('autokakao.ipynb')))\n",
    "\n",
    "# \n",
    "picPosition = pyautogui.locateOnScreen('1.png')\n",
    "print(picPosition)\n",
    "\n",
    "if picPosition is None:\n",
    "    picPosition = pyautogui.locateOnScreen('2.png')\n",
    "    print(picPosition)\n",
    "\n",
    "if picPosition is None:\n",
    "    picPosition = pyautogui.locateOnScreen('3.png')\n",
    "    print(picPosition)\n",
    "\n",
    "clickPosition = pyautogui.center(picPosition)\n",
    "pyautogui.doubleClick(clickPosition)\n",
    "\n",
    "pyperclip.copy(\"테스트용 해위\")\n",
    "pyautogui.hotkey(\"ctrl\", \"v\")\n",
    "time.sleep(1.0)\n",
    "\n",
    "pyautogui.write([\"enter\"])\n",
    "time.sleep(1.0)\n",
    "\n",
    "pyautogui.write([\"escape\"])\n",
    "time.sleep(1.0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "834a2782",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '1.png'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[1;32mIn [4]\u001b[0m, in \u001b[0;36m<cell line: 36>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     33\u001b[0m     pyautogui\u001b[38;5;241m.\u001b[39mwrite([\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mescape\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[0;32m     34\u001b[0m     time\u001b[38;5;241m.\u001b[39msleep(\u001b[38;5;241m1.0\u001b[39m)\n\u001b[1;32m---> 36\u001b[0m \u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[1;32mIn [4]\u001b[0m, in \u001b[0;36msend\u001b[1;34m()\u001b[0m\n\u001b[0;32m      9\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21msend\u001b[39m():\n\u001b[0;32m     10\u001b[0m     threading\u001b[38;5;241m.\u001b[39mTimer(\u001b[38;5;241m10\u001b[39m, send)\u001b[38;5;241m.\u001b[39mstart()\n\u001b[1;32m---> 12\u001b[0m     picPosition \u001b[38;5;241m=\u001b[39m \u001b[43mpyautogui\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlocateOnScreen\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m1.png\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m     13\u001b[0m     \u001b[38;5;28mprint\u001b[39m(picPosition)\n\u001b[0;32m     15\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m picPosition \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py:175\u001b[0m, in \u001b[0;36mraisePyAutoGUIImageNotFoundException.<locals>.wrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    172\u001b[0m \u001b[38;5;129m@functools\u001b[39m\u001b[38;5;241m.\u001b[39mwraps(wrappedFunction)\n\u001b[0;32m    173\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapper\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m    174\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 175\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m wrappedFunction(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    176\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m pyscreeze\u001b[38;5;241m.\u001b[39mImageNotFoundException:\n\u001b[0;32m    177\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ImageNotFoundException\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py:213\u001b[0m, in \u001b[0;36mlocateOnScreen\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    211\u001b[0m \u001b[38;5;129m@raisePyAutoGUIImageNotFoundException\u001b[39m\n\u001b[0;32m    212\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mlocateOnScreen\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m--> 213\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m pyscreeze\u001b[38;5;241m.\u001b[39mlocateOnScreen(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py:373\u001b[0m, in \u001b[0;36mlocateOnScreen\u001b[1;34m(image, minSearchTime, **kwargs)\u001b[0m\n\u001b[0;32m    371\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    372\u001b[0m     screenshotIm \u001b[38;5;241m=\u001b[39m screenshot(region\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;66;03m# the locateAll() function must handle cropping to return accurate coordinates, so don't pass a region here.\u001b[39;00m\n\u001b[1;32m--> 373\u001b[0m     retVal \u001b[38;5;241m=\u001b[39m locate(image, screenshotIm, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    374\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    375\u001b[0m         screenshotIm\u001b[38;5;241m.\u001b[39mfp\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py:353\u001b[0m, in \u001b[0;36mlocate\u001b[1;34m(needleImage, haystackImage, **kwargs)\u001b[0m\n\u001b[0;32m    351\u001b[0m \u001b[38;5;66;03m# Note: The gymnastics in this function is because we want to make sure to exhaust the iterator so that the needle and haystack files are closed in locateAll.\u001b[39;00m\n\u001b[0;32m    352\u001b[0m kwargs[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mlimit\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1\u001b[39m\n\u001b[1;32m--> 353\u001b[0m points \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mtuple\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mlocateAll\u001b[49m\u001b[43m(\u001b[49m\u001b[43mneedleImage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhaystackImage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    354\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(points) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[0;32m    355\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m points[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py:262\u001b[0m, in \u001b[0;36m_locateAll_python\u001b[1;34m(needleImage, haystackImage, grayscale, limit, region, step, confidence)\u001b[0m\n\u001b[0;32m    259\u001b[0m needleFileObj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    260\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(needleImage, (\u001b[38;5;28mstr\u001b[39m, unicode)):\n\u001b[0;32m    261\u001b[0m     \u001b[38;5;66;03m# 'image' is a filename, load the Image object\u001b[39;00m\n\u001b[1;32m--> 262\u001b[0m     needleFileObj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mneedleImage\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mrb\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m    263\u001b[0m     needleImage \u001b[38;5;241m=\u001b[39m Image\u001b[38;5;241m.\u001b[39mopen(needleFileObj)\n\u001b[0;32m    265\u001b[0m haystackFileObj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '1.png'"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-35:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-36:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-37:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-38:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-39:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-40:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-41:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-42:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-43:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-44:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-45:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-46:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-47:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-48:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-49:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-50:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-51:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-52:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-53:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-54:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-55:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-56:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-57:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-58:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-59:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-60:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n",
      "Exception in thread Thread-61:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 973, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\threading.py\", line 1299, in run\n",
      "    self.function(*self.args, **self.kwargs)\n",
      "  File \"C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_9648\\4207577363.py\", line 12, in send\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 175, in wrapper\n",
      "    return wrappedFunction(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyautogui\\__init__.py\", line 213, in locateOnScreen\n",
      "    return pyscreeze.locateOnScreen(*args, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 373, in locateOnScreen\n",
      "    retVal = locate(image, screenshotIm, **kwargs)\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 353, in locate\n",
      "    points = tuple(locateAll(needleImage, haystackImage, **kwargs))\n",
      "  File \"C:\\Users\\user\\anaconda3\\lib\\site-packages\\pyscreeze\\__init__.py\", line 262, in _locateAll_python\n",
      "    needleFileObj = open(needleImage, 'rb')\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '1.png'\n"
     ]
    }
   ],
   "source": [
    "import pyautogui\n",
    "import pyperclip\n",
    "import time\n",
    "import threading\n",
    "import os\n",
    "\n",
    "os.chdir(os.path.dirname(os.path.abspath('autokakao.ipynb')))\n",
    "\n",
    "def send():\n",
    "    threading.Timer(10, send).start()\n",
    "    \n",
    "    picPosition = pyautogui.locateOnScreen('1.png')\n",
    "    print(picPosition)\n",
    "    \n",
    "    if picPosition is None:\n",
    "        picPosition = pyautogui.locateOnScreen('2.png')\n",
    "        print(picPosition)\n",
    "    \n",
    "    if picPosition is None:\n",
    "        picPosition = pyautogui.locateOnScreen('3.png')\n",
    "        print(picPosition)\n",
    "    \n",
    "    clickPosition = pyautogui.center(picPosition)\n",
    "    pyautogui.doubleClick(clickPosition)\n",
    "    \n",
    "    pyperclip.copy(\"세번째 테스트 내이름 부르면 해위~\")\n",
    "    pyautogui.hotkey(\"ctrl\", \"v\")\n",
    "    time.sleep(1.0)\n",
    "    \n",
    "    pyautogui.write([\"enter\"])\n",
    "    time.sleep(1.0)\n",
    "    \n",
    "    pyautogui.write([\"escape\"])\n",
    "    time.sleep(1.0)\n",
    "\n",
    "send()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "09402747",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'left'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[1;32mIn [4]\u001b[0m, in \u001b[0;36m<cell line: 14>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     11\u001b[0m \u001b[38;5;28mprint\u001b[39m(picPosition)\n\u001b[0;32m     13\u001b[0m \u001b[38;5;66;03m# 사진의 좌표를 찍어주고 좌표값은 모니터 기준\u001b[39;00m\n\u001b[1;32m---> 14\u001b[0m \u001b[43mpicPosition\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mleft\u001b[49m\n\u001b[0;32m     16\u001b[0m \u001b[38;5;66;03m# 마우스를 이동시켜준다\u001b[39;00m\n\u001b[0;32m     17\u001b[0m \u001b[38;5;66;03m# 사진 기준으로 0,0으로 이동\u001b[39;00m\n\u001b[0;32m     18\u001b[0m pyautogui\u001b[38;5;241m.\u001b[39mmoveTo(picPosition\u001b[38;5;241m.\u001b[39mleft, picPosition\u001b[38;5;241m.\u001b[39mtop)\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'left'"
     ]
    }
   ],
   "source": [
    "# fps 핵 만들기\n",
    "# https://simritest.com/reaction/\n",
    "import pyautogui\n",
    "import pyperclip\n",
    "import time\n",
    "import os\n",
    "\n",
    "# 경로를 .py파일의 실행경로로 이동, 현재 경로로 이동\n",
    "os.chdir(os.path.dirname(os.path.abspath('autokakao.ipynb')))\n",
    "\n",
    "n = 0\n",
    "while(n<5):\n",
    "    # 모니터 기준 사진 좌표, 가로길이 세로길이를 찍어준다.\n",
    "    # 픽셀수가 적고, 단색의 이미지일수록 더 빠르다.\n",
    "    # 모니터 기준 사진 좌표, 가로길이 세로길이를 찍어준다.\n",
    "    picPosition = pyautogui.locateOnScreen('click1.png')\n",
    "    print(picPosition)\n",
    "    if picPosition != None:\n",
    "        # 마우스를 이동시켜준다(picPosition.left는 좌우, top은 상하)\n",
    "        # 사진 기준으로 0,0으로 이동\n",
    "        # 사진이 없으면 좌표를 못찾아서 오류가 난다.\n",
    "        pyautogui.moveTo(picPosition.left, picPosition.top)\n",
    "        \n",
    "        # 클릭한다. default 값은 좌클릭\n",
    "        # pyautogui.click()\n",
    "        pyautogui.click()\n",
    "        n += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "035c4c5d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1018, top=301, width=7, height=7)\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1018, top=301, width=7, height=7)\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1018, top=301, width=7, height=7)\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1018, top=301, width=7, height=7)\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "Box(left=1018, top=301, width=7, height=7)\n"
     ]
    }
   ],
   "source": [
    "# fps 핵 만들기\n",
    "# https://simritest.com/reaction/\n",
    "import pyautogui\n",
    "import pyperclip\n",
    "import time\n",
    "import os\n",
    "\n",
    "# 경로를 .py파일의 실행경로로 이동, 현재 경로로 이동\n",
    "os.chdir(os.path.dirname(os.path.abspath('autokakao.ipynb')))\n",
    "\n",
    "n = 0\n",
    "while(n<5):\n",
    "    picPosition = pyautogui.locateOnScreen('click1.png')\n",
    "    print(picPosition)\n",
    "    if picPosition != None:\n",
    "        pyautogui.click(picPosition.left, picPosition.top)\n",
    "        n += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba8182a7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
